buffer/src/main/java/io/netty/buffer/PoolArena.java

```diff
@@ -80,12 +82,15 @@
     // TODO: Test if adding padding helps under contention
     //private long pad0, pad1, pad2, pad3, pad4, pad5, pad6, pad7;
 
-    protected PoolArena(PooledByteBufAllocator parent, int pageSize, int maxOrder, int pageShifts, int chunkSize) {
+    protected PoolArena(PooledByteBufAllocator parent, int pageSize,
+          int maxOrder, int pageShifts, int chunkSize, int cacheAlignment) {
         this.parent = parent;
         this.pageSize = pageSize;
         this.maxOrder = maxOrder;
         this.pageShifts = pageShifts;
         this.chunkSize = chunkSize;
+        this.directMemoryCacheAlignment = cacheAlignment;
```

Comment:
does this have to be a power of 2 for the `mask` below to work? if so should we enforce that somewhere for example: warn and go to the next positive power of 2, or set to 0?

`MathUtil.safeFindNextPositivePowerOfTwo` maybe useful here.